Core/clim-basic/input.lisp

```diff
@@ -288,79 +298,71 @@ use condition-variables nor locks."))
 (defmethod event-queue-peek ((queue simple-event-queue))
   (do-port-force-output queue)
   (check-schedule queue)
-  (or (first (event-queue-head queue))
-      (when (process-next-event (event-queue-port queue) :timeout 0)
-        (first (event-queue-head queue)))))
+  (process-next-event (event-queue-port queue) :timeout 0)
```

Comment:
it is deliberate choice. in concurrent-event-queue we do not call process-next-event at all (because it runs on its own thread).

This unconditional call is made to ensure event processing progress (i.e we may have a wait function which yields true only after some event is processed by the port, not event queued - like selection-request events).

Also assuming that scheduler appends events as we go, we may never feel a need to call process-next-event causing mouse or keyboard events being dropped.